src/blk/kernel/KernelDevice.cc

```diff
@@ -414,9 +414,15 @@ bool KernelDevice::get_thin_utilization(uint64_t *total, uint64_t *avail) const
 
 int KernelDevice::choose_fd(bool buffered, int write_hint) const
 {
-  assert(write_hint >= WRITE_LIFE_NOT_SET && write_hint < WRITE_LIFE_MAX);
+#if defined(F_SET_FILE_RW_HINT)
```

Comment:
@wjwithagen could you please extract this change into a separate commit in this changeset?

Reply:
Fixed